src/server/models/okta/Authentication.ts

```diff
@@ -31,60 +44,79 @@ enum AuthenticationFactorResult {
 }
 
 // https://developer.okta.com/docs/reference/api/authn/#links-object
-interface AuthenticationLinks {
-  next?: HalLink;
-  prev?: HalLink;
-  cancel?: HalLink;
-  skip?: HalLink;
-  resend?: HalLink;
-}
-
-// https://developer.okta.com/docs/reference/api/authn/#user-object
-interface AuthenticationEmbeddedUser {
-  id: string;
-  passwordChanged?: string;
-  profile: AuthenticationEmbeddedUserProfile;
-  recovery_question?: AuthenticationEmbeddedRecoveryQuestion;
-}
+const authenticationLinksSchema = z.object({
+  next: halLinkSchema.nullable().optional(),
+  prev: halLinkSchema.nullable().optional(),
+  cancel: halLinkSchema.nullable().optional(),
+  skip: halLinkSchema.nullable().optional(),
+  resend: halLinkSchema.nullable().optional(),
+});
 
 // https://developer.okta.com/docs/reference/api/authn/#user-profile-object
-interface AuthenticationEmbeddedUserProfile {
-  firstName: string;
-  secondName: string;
-  locale?: string;
-  login: string;
-  timeZone?: string;
-}
+const authenticationEmbeddedUserProfileSchema = z.object({
+  login: z.string(),
+  // okta says this is not nullable or optional, but there are some
+  // users in our system that do not have a firstName/lastName or it is null
+  firstName: z.string().nullable().optional(),
```

Comment:
we also use `.nullable()` with `.optional()` everywhere, as in an Okta API response, the response could either be `null` or not defined, hence we need both to handle these cases